#! /bin/sh /usr/share/dpatch/dpatch-run
## fix-exit-status.dpatch by  <matthijs@monster.cacholong.nl>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix exit status

@DPATCH@
diff -urNad trunk/pdns/pdns_recursor.cc /tmp/dpep.Pgcny3/trunk/pdns/pdns_recursor.cc
--- trunk/pdns/pdns_recursor.cc	2005-04-17 10:59:50.949753728 +0200
+++ /tmp/dpep.Pgcny3/trunk/pdns/pdns_recursor.cc	2005-05-04 18:24:17.257315824 +0200
@@ -395,6 +395,7 @@
 
 int main(int argc, char **argv) 
 {
+  int ret = EXIT_SUCCESS;
 #ifdef WIN32
     WSADATA wsaData;
     WSAStartup( MAKEWORD( 2, 0 ), &wsaData );
@@ -623,17 +624,20 @@
   }
   catch(AhuException &ae) {
     L<<Logger::Error<<"Exception: "<<ae.reason<<endl;
+    ret=EXIT_FAILURE;
   }
   catch(exception &e) {
     L<<Logger::Error<<"STL Exception: "<<e.what()<<endl;
+    ret=EXIT_FAILURE;
   }
   catch(...) {
     L<<Logger::Error<<"any other exception in main: "<<endl;
+    ret=EXIT_FAILURE;
   }
   
 #ifdef WIN32
   WSACleanup();
 #endif // WIN32
 
-  return 0;
+  return ret;
 }
